frontend/src/composables/errorService.ts

```diff
@@ -0,0 +1,36 @@
+import {reactive} from 'vue';
+import {AxiosError} from "axios";
+import {useToastService} from "@/composables/toastService";
+
+export function useErrorService() {
+
+  const errorState = reactive({
+    errorCode: '',
+    errorMessage: '',
+    // some other properties...
+  });
+
+  const handleError = (error: AxiosError | any, message?: string) => {
+    errorState.errorCode = error.isAxiosError ? error.response.data.errorCode : 'Unknown Error';
+    errorState.errorMessage = error.isAxiosError ? error.response.data.message : message || 'Service Unavailable';
+    return errorState;
+  }
+
+  const handleAndThrow = (error: AxiosError | any, message?: string) => {
+    const newError = handleError(error, message);
+    useToastService().showError('ERROR', `<b>${newError.errorCode}</b><br>${newError.errorMessage}`);
+    throw new Error(Object.values(newError).filter(i => i).join(': '), {cause: error});
```

Comment:
Does this `.filter(i => i)` do anything?

Reply:
yep, it's filtering null values :)
as of now that filtering can be omitted because I am passing default values if no values for error code or message are provided. But, if we will add some other, mysterious properties for the Error object then it would be needed :)